main.playground/Contents.swift

```diff
@@ -1,26 +1,121 @@
 class Fruit {
-    static var id = 0;
-    var name: String;
-    var weight: Float; // grams
-    var price: Float; // cents
-    var id: Int;
-    
-    init(name: String, weight: Float, price: Float) {
-        Fruit.id = Fruit.id + 1;
-        self.id = Fruit.id;
-        self.name = name;
-        self.weight = weight;
-        self.price = price;
+    
+    static var id: Int = 0
+    var id: Int
+    var type: String
+    // Weight in grams
+    var weight: Float
+    
+    init(weight: Float, type: String) {
+        Fruit.id = Fruit.id + 1
+        self.id = Fruit.id
+        self.type = type
+        self.weight = weight
+    }
+    
+    func toString() -> String {
+        return "ID: \(self.id), Type: \(self.type), Weight: \(self.weight)gr"
+    }
+}
+
+class Greengrocer {
+    
+    var fruits:[String: [Fruit]] = [:]
+    // var fruits2:[String: (Float, [Fruit])] = [:]
+    var prices:[String:Float] = [:]
+    
+    // Potremmo inizializzare la classe con il valore del prezzo/kg di ogni singolo frutto
+    // passando come parametro un dictionary come -prices-, ma per semplicità usiamo un approccio diverso
```

Comment:
Eliminiamo i commenti inutili, non abbiamo property che necessitano di essere inizializzate in questa classe.